### ISSUE TITLE
Possibly out of date error in autograd codegen

### ISSUE THREAD

**Issue by soulitzer** [pytorch/pytorch#69714]
We update the codegen in the ADInplaceOrView kernel now, so the error can probably be removed

@soulitzer what's the best way to declare these ops as non-differentiable? I guess the best case scenario would be if we can error out immediately in the forward pass, if someone tries to call `torch.logical_and(torch.ones(2, requres_grad=True))`.

I'm getting codegen errors that say `RuntimeError: output_differentiability=False for inplace operation (version_counter won't get updated)`. I see another option from the docs is to add these ops to the `DONT_REQUIRE_DERIVATIVE` in `gen_variable_type`?

_Originally posted by @bdhirsh in https://github.com/pytorch/pytorch/pull/69169#discussion_r760198800_

**Comment by albanD** [comment]
These non-differentiable ops should definitely not raise an error. Only the output should not require gradients.
And if this is the inplace version of these ops, then the gradient should be 0.

**Comment by soulitzer** [comment]
Agreed, this is a different issue though. I edited the description to clarify.
